src/treelearner/cuda_kernel_launcher.cu

```diff
@@ -38,41 +38,41 @@ void cuda_histogram(
     if (leaf_num_data == num_data) {
       if (use_all_features) {
         if (!is_constant_hessian)
-          histogram16<<<16*num_workgroups, 16, 0, stream>>>(arg0, arg1, arg2,
+          histogram16<<<num_workgroups, 16, 0, stream>>>(arg0, arg1, arg2,
```

Comment:
Perhaps a better fix would be to change how the `group_id` is calculated in the kernel, instead of simple remove of the multiplications. I need to perform some evaluation to see how the efficiency changes with fewer number of blocks.